lib/cocoapods-core/source/aggregate.rb

```diff
@@ -50,6 +50,7 @@ def all_sets
       # @return [Set] The most representative set for the Pod with the given
       #         name. Returns nil if no representative source found containing a pod with given name.
       #
+      # Deprecated: Please use `representative_set_for_search` instead.
```

Comment:
@yanzhiwei147 can you explain a bit why this needs to be deprecated and not just removed?

Reply:
It can used on other scenes, so mark deprecated. But it can removed also.

Reply:
Please add an explanation of _why_ it is deprecated in the deprecated message

Reply:
What can i remove the function instead of comment deprecated?